### ISSUE TITLE
[1.14.4] Map Leave Pearl Possibly Causing Nether Trouble

### ISSUE THREAD

**Issue by Warlok333** [RobertSkalko/Mine-and-Slash#132]
While playing "World of Dragons - A New Age (WODNA)" modpack version 1.35.36 on the public, official server, I came across an issue that I believe is caused by the "Map Leave Pearl" in my inventory while leaving the Nether. The first time I entered the Nether, I acquired a Map Leave Pearl (I think it was just on the ground). When I died (set on fire by lava), I was automatically teleported near Spawn (not my bed), near the surface, in water. I then used "/back" to return and recover any lost gear, was set on fire again and died away from the flowing lava, trying to prevent further gear loss. This time, however, I was teleported to the bottom of a lava lake near Spawn and lost everything. After gearing back up and heading back into the Nether, I had to log out while still in the Nether. When I logged back in, it was back to the bottom of the lava lake. I again geared up and went into the Nether but tried to exit the Nether through the portal. This time, a new portal was created in the Overworld about 300 blocks away from my original portal, near again to the Spawn area and the lava lake. I went back through this portal and arrived at the correct portal in the Nether, then tried using a Map Leave Pearl which worked by teleporting me to the water near Spawn just like the first time I died.

I cannot be 100% certain I had a Map Leave Pearl in my inventory every single time, but at least 3 of those times I either had one in my inventory or used one to leave the Nether. If there is additional information I can provide, I am happy to do so.

**Comment by Warlok333** [comment]
Confirmed every time I enter the Nether, I receive a Map Leave Pearl with the exact same stats. I have several stacked in a chest, now.

**Comment by RobertSkalko** [comment]
This is probably both modpack's fault and mine.

First, he set nether as a map in config, which is why you got the leave pearl there.

Second, my problem because i shouldn't be giving out leave pearls in dimensions that aren't from my mod. Cus other dimensions have ways to get out. I also shouldn't teleport players out of config added map dimensions on death. 

So these will be fixed. But it has nothing to do with the pearl, it's not bugging things or anything :p 

The only uncertainty is if i'll fix the death in maps teleporting back to spawn instead of near map device. This is a long bug i've been trying to fix for a while..

Closing, @AzureDoom  if you take up bugfixing for 1.14 https://github.com/RobertSkalko/Mine-and-Slash/commit/a85812e1f0242f4223cf51b7b69d8d8218b70fb5

**Comment by kreezxil** [comment]
@RobertSkalko WODNA is my pack. here in this section of the config is the MAP LEAVE PEARL?

```json
    "minecraft:the_nether": {
      "MOB_LEVEL_PER_DISTANCE": 75,
      "MOB_LEVEL_ONE_AREA": 25,
      "MAXIMUM_MOB_LEVEL": 1000,
    "MINIMUM_MOB_LEVEL": 1,
    "LEVEL_FOR_MOBS_TO_BE_LEGENDARY": 200,
    "LEVEL_FOR_MOBS_TO_BE_MYTHICAL": 800,
    "SCALE_MOB_LEVEL_TO_NEAREST_PLAYER": false,
      "DROPS_UNIQUE_ITEMS": true,
      "MAP_TIER": 4,
      "DROP_MULTIPLIER": 1.2,
      "MOB_STRENGTH_MULTIPLIER": 1.3
    },
```

**Comment by RobertSkalko** [comment]
MAP_TIER

If higher than 0 or 1,  it counts as a map dimension currently. It'll be fixed for 1.15 but for now you can just set it to 0 if it's a problem

**Comment by kreezxil** [comment]
yeah, ok, thanks. it's a problem right now, so i'll adjust it downward in
the config. If AZUREDOOM could patch the 1.14.4 version with your 1.15.2
changes that would be great!
